Models for PublicRepository and Accessions

ttung opened the pull request:
### Description
Pretty boiler plate stuff. :)

Depends on #58, #59

#### Issue
[ch64743](https://app.clubhouse.io/genepi/stories/space/64743)

### Test plan
Ran the migration.

erinstargazer commented on src/py/aspen/database/models/accessions.py:
```diff
@@ -0,0 +1,87 @@
+_PublicRepositoryTypeTable = enumtables.EnumTable(
+    PublicRepositoryType,
+    base,
+    tablename="publicrepositorytypes",
```
What's the naming convention? I would have guessed this would be `public_repository_types`. Are we doing something special for the enum tables with the names?

ttung replied:
I can change it.

erinstargazer reviewed (approved):
I have a question about the naming conventions but it looks consistent in the code base so I'm approving.

Commit:
Models for PublicRepository and Accessions

### Description
Pretty boiler plate stuff. :)

#### Issue
[ch64743](https://app.clubhouse.io/genepi/stories/space/64743)

### Test plan
Ran the migration.
